catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -375,78 +359,42 @@ const useFileStyles = M.makeStyles((t) => ({
   },
 }))
 
-type FileProps = tagged.ValueOf<typeof FilesEntry.File> & {
-  prefix?: string
-  dispatch: DispatchFilesAction
+interface FileOwnProps {
+  name: string
+  state?: FilesEntryState
+  size?: number
+  action?: React.ReactNode
 }
 
-function File({ disabled, name, type, size, prefix, dispatch }: FileProps) {
-  const classes = useFileStyles()
-
-  const path = (prefix || '') + name
-
-  const handle = React.useCallback(
-    (cons: tagged.ConstructorOf<typeof FilesAction>) => (e: React.MouseEvent) => {
-      e.stopPropagation()
-      dispatch(cons(path))
-    },
-    [dispatch, path],
-  )
-
-  const action = React.useMemo(
-    () =>
-      ({
-        added: { hint: 'Remove', icon: 'clear', handler: handle(FilesAction.Revert) },
-        modified: {
-          hint: 'Revert',
-          icon: 'undo',
-          handler: handle(FilesAction.Revert),
-        },
-        hashing: {
-          hint: 'Revert',
-          icon: 'undo',
-          handler: handle(FilesAction.Revert),
-        },
-        deleted: {
-          hint: 'Restore',
-          icon: 'undo',
-          handler: handle(FilesAction.Revert),
-        },
-        unchanged: { hint: 'Delete', icon: 'clear', handler: handle(FilesAction.Delete) },
-      }[type]),
-    [type, handle],
-  )
+type FileProps = FileOwnProps &
+  Omit<React.HTMLAttributes<HTMLDivElement>, keyof FileOwnProps>
```

Comment:
even this

```tsx
interface FileProps extends React.HTMLAttributes<HTMLDivElement> {
  name: string
  state?: FilesEntryState
  size?: number
  action?: React.ReactNode
}
```

Reply:
the point of this is to exclude props that are not passed to the div from the type to avoid possible conflicts and to articulate that those props arent related to the underlying html element

Reply:
I wrote the same: "use HTMLAttributes from DivElement but re-define name, state size and action", didn't?

Reply:
i was about to say that you cant override a prop of an interface with incompatible type, but seems like i was confusing declaration merging with plain interface extension, so you're probably right